storage/memory/memory.go

```diff
@@ -456,9 +459,17 @@ func (m *memory) Objects(ctx context.Context, s *node.Node, p *predicate.Predica
 		}
 	}
 
-	for _, t := range selectedTrpls {
-		if t != nil && ckr.CheckLimitAndUpdate() {
-			objs <- t.Object()
+	st := make(map[string]*triple.Triple)
+	var strObs []string
+	for _, k := range selectedTrpls {
+		st[k.String()] = k
+		strObs = append(strObs, k.String())
+	}
+	sort.Strings(strObs)
```

Comment:
Does it make sense to extract this to a function?

Reply:
+1

Reply:
Done